### DISCUSSION TITLE
Add support for additional extension chunks for ICC and XMP in gifload

### DISCUSSION THREAD

**Discussion by jfcalvo** [libvips/libvips#2064] [Ideas]
As discussed here https://github.com/libvips/libvips/issues/1722#issuecomment-660486695 looks like `vipsheader` command can't detect or extract the ICC profile attached to the following `gif` file:

![example](https://user-images.githubusercontent.com/446497/87855794-e239e880-c91a-11ea-8f1f-582effdeb99e.gif)

Even with this file correctly showing a ICC Profile `Adobe RGB (1998)` on MacOS Preview is not possible to get that information using `vipsheader`:

```
$ vipsheader -a example.gif
example.gif: 1000x667 uchar, 3 bands, srgb, gifload
width: 1000
height: 667
bands: 3
format: uchar
coding: none
interpretation: srgb
xoffset: 0
yoffset: 0
xres: 1
yres: 1
filename: example.gif
vips-loader: gifload
n-pages: 1
loop: 1
delay: 0
```

It's not possible to dump the `icc-profile-data` too:

```
$ vipsheader -f icc-profile-data example.gif
vipsheader: vips_image_get: field "icc-profile-data" not found
```

According to this code looks like there are extension chunks for ICC and XMP for gif format:

https://chromium.googlesource.com/webm/libwebp/+/v0.3.0-rc6/examples/gif2webp.c#387

libvips supports the NETSCAPE and ANIM chunks, but not the other two. 

The idea of this issue is to add support for these two other extensions so the library is capable of managing and works with them.

`vipsheader` should also be capable of read and extract the `icc-profile-data` for this gif file too.

**Comment by jfcalvo** [comment]
@jcupitt please feel free to edit the definition of the issue with more information or some clarification. Thanks.

**Comment by jcupitt** [comment]
Looks good, thanks!
